test/test_data.py

```diff
@@ -0,0 +1,57 @@
+import numpy as np
+
+from carla.data.catalog import DataCatalog
+
+
+def test_adult_col():
+    data_name = "adult"
+    data_catalog = "adult_catalog.yaml"
+    data_catalog = DataCatalog(data_name, data_catalog)
+
+    actual_col = (
+        data_catalog.categoricals + data_catalog.continous + [data_catalog.target]
+    )
+    actual_col = actual_col.sort()
+    expected_col = data_catalog.raw.columns.values
+    expected_col = expected_col.sort()
+
+    assert actual_col == expected_col
+
+
+def test_adult_norm():
+    data_name = "adult"
+    data_catalog = "adult_catalog.yaml"
+    data_catalog = DataCatalog(data_name, data_catalog)
+
+    col = data_catalog.continous
+
+    raw = data_catalog.raw[col]
+    norm = data_catalog.normalized[col]
+
+    assert ((raw != norm).all()).any()
+
+
+def test_adult_enc():
+    data_name = "adult"
+    data_catalog = "adult_catalog.yaml"
+    data_catalog = DataCatalog(data_name, data_catalog)
+    cat = data_catalog.encoded
+
+    assert cat.select_dtypes(exclude=[np.number]).empty
+
+
+def test_adult_norm_enc():
+    data_name = "adult"
+    data_catalog = "adult_catalog.yaml"
+    data_catalog = DataCatalog(data_name, data_catalog)
+
+    norm_col = data_catalog.continous
+    norm_enc_col = data_catalog.encoded_normalized.columns
+
+    cat = data_catalog.encoded.copy()
```

Comment:
Love the tests, way to go 💪🏽 

Side note here, you noticed a bad design in the datacatalog where the data retrieved is not immutable (which it should be. I.e. we don't want the data _inside_ the catalog change if we change the returned object)

I was working on a different feature, but then never pushed the fix, if you want you can include it in this PR, so you don't need it `.copy()` workaround here

```
     @property
     def raw(self):
-        return self._raw
+        return self._raw.copy()
 
     @property
     def normalized(self):
         if self._data_normalized is None:
             self._data_normalized = processing.normalize(self.raw, self.continous)
 
-        return self._data_normalized
+        return self._data_normalized.copy()
 
     @property
     def encoded(self):
         if self._data_encoded is None:
             self._data_encoded = pd.get_dummies(self.raw, drop_first=True)
 
-        return self._data_encoded
+        return self._data_encoded.copy()
 
     @property
     def encoded_normalized(self):
@@ -57,7 +57,7 @@ class DataCatalog:
                 self.encoded, self.continous
             )
 
-        return self._data_encoded_normalized
+        return self._data_encoded_normalized.copy()
```

Reply:
Great, will include this in this PR